File: backend/src/race_command_center/models/part.py
from pydantic import BaseModel, Field


class CircuitSpecificPart(BaseModel):
    part_id: str
    name: str
    part_type: str
    target_circuit_id: str

    problem_statement: str
    technical_hypothesis: str
    expected_impact: str

    material: str | None = None
    estimated_weight_g: float | None = None
    manufacturing_method: str | None = None

    risk_level: str = "medium"
    status: str = "concept"

    simulation_id: str | None = None
    evidence: list[dict] = Field(default_factory=list)
    approval_status: str = "pending"
    metadata: dict = Field(default_factory=dict)


class PartCreate(BaseModel):
    name: str
    part_type: str
    target_circuit_id: str
    problem_statement: str
    technical_hypothesis: str
    expected_impact: str
    material: str | None = None
    estimated_weight_g: float | None = None
    manufacturing_method: str | None = None
    risk_level: str = "medium"

File: frontend/src/types/part.ts
export type PartStatus =
  | "concept"
  | "designed"
  | "simulated"
  | "approved_for_manufacturing"
  | "manufactured"
  | "mounted"
  | "tested"
  | "rejected"
  | "archived";

export type RiskLevel = "low" | "medium" | "high" | "critical";

export type CircuitSpecificPart = {
  part_id: string;
  name: string;
  part_type: string;
  target_circuit_id: string;
  problem_statement: string;
  technical_hypothesis: string;
  expected_impact: string;
  material?: string;
  estimated_weight_g?: number;
  manufacturing_method?: string;
  risk_level: RiskLevel;
  status: PartStatus;
  simulation_id?: string;
  evidence: Record<string, unknown>[];
  approval_status: string;
  metadata: Record<string, unknown>;
};
